core/src/main/java/org/jboss/pnc/build/finder/protobuf/ListKojiArchiveInfoProtobufWrapper.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.build.finder.protobuf;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+
+import com.redhat.red.build.koji.model.xmlrpc.KojiArchiveInfo;
+
+/**
+ * Class to wrap around a List of KojiArchiveInfo. This is used so that Protostream can properly marshall / unmarshall
+ * the list and avoid this Protostream bug (https://issues.redhat.com/browse/IPROTO-219).
+ */
+public class ListKojiArchiveInfoProtobufWrapper {
+    /**
+     * We cannot use Collections.emptyList because Protostream has no idea how to marshall it. Instead, we use an empty
+     * ArrayList.
+     */
+    private List<KojiArchiveInfo> data = new ArrayList<>(0);
```

Comment:
Something doesn't seem right. You set `data` here. but then it gets overwritten in the constructor.

If I understand what you are trying to do, we could make this `private static final EMPTY_LIST` and then:

```
public List<KojiArchiveInfo> getData() {
     return data != null ? data : EMPTY_LIST;
}
```

Reply:
Sure done